vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/ComponentWrap.java

```diff
@@ -75,14 +76,36 @@ private boolean isEffectivelyVisible(Component component) {
      * Automatically generates a client side change to propagate modality.
      *
      * @param modal
-     *         {@code true} to make component modal, {@code false} to remove
-     *         modality
+     *            {@code true} to make component modal, {@code false} to remove
+     *            modality
      */
     public void setModal(boolean modal) {
         UI.getCurrent().setChildComponentModal(component, modal);
+        flushChanges();
+    }
 
+    /**
+     * Checks that wrapped component is usable, otherwise throws an
+     * {@link IllegalStateException} with details on the current state of the
+     * component.
+     */
+    protected final void ensureComponentIsUsable() {
+        if (!isUsable()) {
+            throw new IllegalStateException(
+                    PrettyPrintTreeKt.toPrettyString(component)
+                            + " is not usable");
+        }
+    }
+
+    /**
+     * Simulates a server round-trip, flushing pending component changes.
+     */
+    static void flushChanges() {
```

Comment:
This might need to be protected as the component wrappers when moved to components repository might end up under `com/vaadin/flow/component/[componentName]/testbench/` like the elements.

Also this could almost be named `roundTrip` and be added to the BaseUnitTest class so that users could also use this when a simulated roundtrip would be needed.

Reply:
Renamed and added the method to BaseUnitTest. ComponentWrap also has a protected `roundTrip` method that delegates to BaseUnitTest. Hope I correctly understood your comment.